content/blog/2025-07-01-datafusion-comet-0.9.0.md

```diff
@@ -0,0 +1,176 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+The Apache DataFusion PMC is pleased to announce version 0.9.0 of the [Comet](https://datafusion.apache.org/comet/) subproject.
+
+Comet is an accelerator for Apache Spark that translates Spark physical plans to DataFusion physical plans for
+improved performance and efficiency without requiring any code changes.
+
+This release covers approximately ten weeks of development work and is the result of merging 139 PRs from 24
+contributors. See the [change log] for more information.
+
+[change log]: https://github.com/apache/datafusion-comet/blob/main/dev/changelog/0.9.0.md
+
+## Release Highlights
+
+### Complex Type Support in Parquet Scans
+
+Comet now supports complex types (Structs, Maps, and Arrays) when reading Parquet files. This functionality is not
+yet available when reading Parquet files from Apache Iceberg.
+
+This functionality was only available in previous releases when manually specifying one of the new experimental
+scan implementations. Comet now automatically chooses the best scan implementation based on the input schema, and no
+longer requires manual configuration.
+
+### Complex Type Processing Improvements
+
+Numerous improvements have been made to complex type support to ensure Spark-compatible behavior when casting between
+structs and accessing fields within deeply nested types.
+
+### Shuffle Improvements
+
+Comet now accelerates a broader range of shuffle operations, leading to more queries running fully natively. In
+previous releases, some shuffle operations fell back to Spark to avoid some known bugs in Comet, and these bugs have
+now been fixed.
+
+### New Features
+
+Comet 0.9.0 adds support for the following Spark expressions:
+
+- ArrayDistinct
+- ArrayMax
+- ArrayRepeat
+- ArrayUnion
+- BitCount
+- BitNot
+- Expm1
+- MapValues
+- Signum
+- ToPrettyString
+- map[]
+
+### Improved Spark SQL Test Coverage
+
+Comet now passes 97% of the Spark SQL test suite, with more than 24,000 tests passing (based on testing against
+Spark 3.5.6).
+
+This release contains numerous bug fixes to achieve this coverage, including improved support for exchange reuse
+when AQE is enabled. The remaining ignored tests are mostly related to metric differences or tests irrelevant to
+Comet, such as tests for whole-stage code generation.
+
+<style>
+  table {
+    border-collapse: collapse;
+    width: 100%;
+    font-family: sans-serif;
+  }
+
+  th, td {
+    text-align: left;
+    padding: 8px 12px;
+  }
+
+  th {
+    background-color: #f2f2f2;
+    font-weight: bold;
+  }
+
+  td {
+    border-bottom: 1px solid #ddd;
+  }
+
+  tbody tr:last-child td {
+    font-weight: bold;
+    border-top: 2px solid #000;
+  }
+</style>
+
+<table>
+  <thead>
+    <tr>
+      <th>Module</th>
```

Comment:
Wondering what `Module` means here...

Reply:
This is the header for the column containing values `core-1`, `core-2`, etc